src/axom/sidre/examples/sidre_external_array.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+// Axom includes
+#include "axom/core.hpp"  // for Axom types and macros
+#include "axom/sidre.hpp" // for sidre
+#include "axom/slic.hpp"  // for logging with slic
+
+// MPI includes
+#include <mpi.h>
+
+// C/C++ includes
+#include <iostream>
+
+// aliases
+namespace sidre = axom::sidre;
+namespace slic  = axom::slic;
+
+//------------------------------------------------------------------------------
+void sidre_write( MPI_Comm comm,
+                  const std::string& file,
+                  int* data,
+                  axom::IndexType numTuples,
+                  axom::IndexType numComponents )
+{
+  SLIC_ASSERT( comm != MPI_COMM_NULL );
+  SLIC_ASSERT( data != nullptr );
+  SLIC_ASSERT( !file.empty( ) );
+
+  int nranks = -1;
+  MPI_Comm_size( comm, &nranks );
+
+  sidre::DataStore ds;
+  sidre::Group* root = ds.getRoot();
+  sidre::View*  view = root->createView( "data" );
+
+  sidre::IndexType shape[2];
+  shape[ 0 ] = numTuples;
+  shape[ 1 ] = numComponents;
+
+  view->setExternalDataPtr( sidre::INT32_ID, 2, shape, data );
+
+// DEBUG
+  SLIC_INFO( "Here is the data that is begin dumped:" );
+  root->print();
+  std::cout << std::endl;
+// DEBUG
+
+
+  // STEP 2: save the array data in to a file
+  sidre::IOManager sidre_io( comm );
+  sidre_io.write( root, nranks, file, "sidre_hdf5" );
+}
+
+//------------------------------------------------------------------------------
+void sidre_read( MPI_Comm comm,
+                 const std::string& file,
+                 int*& data,
+                 axom::IndexType& numTuples,
+                 axom::IndexType& numComponents )
+{
+  SLIC_ASSERT( comm != MPI_COMM_NULL );
+  SLIC_ASSERT( data == nullptr );
+  SLIC_ASSERT( !file.empty( ) );
+
+  int nranks = -1;
+  MPI_Comm_size( comm, &nranks );
+
+  sidre::DataStore ds;
+  sidre::Group* root = ds.getRoot();
+
+  sidre::IOManager sidre_io( comm );
+  sidre_io.read( root, file );
+
+  SLIC_ASSERT( root->hasChildView("data") );
+  sidre::View* view = root->getView( "data" );
+
+  sidre::IndexType shape[2];
+  view->getShape( 2, shape );
+  numTuples     = shape[ 0 ];
+  numComponents = shape[ 1 ];
+
+  axom::IndexType nelems = view->getNumElements();
+  SLIC_ASSERT( nelems==(numTuples*numComponents) );
+
+  data = axom::allocate< int >(nelems);
+  SLIC_ASSERT( data != nullptr );
+  SLIC_ASSERT( view->isDescribed() );
+  SLIC_ASSERT( view->isExternal() );
+
+  view->setExternalDataPtr( data );
```

Comment:
Since this was not obvious to you when developing it, please add some comments.
E.g. something like
> // Inform Sidre of the locations to the external data 

and 

> // load the data into these pointers

Reply:
Yeah, definitely not obvious -- I'll add some comments.